src/platformdirs/windows.py

```diff
@@ -0,0 +1,170 @@
+import ctypes
+import os
+import sys
+from functools import lru_cache
+from typing import Callable, Optional
+
+from .api import PlatformDirsABC
+
+
+class Windows(PlatformDirsABC):
+    """`MSDN on where to store app data files
+    <http://support.microsoft.com/default.aspx?scid=kb;en-us;310294#XSLTH3194121123120121120120>`_.
+    Makes use of the
+    `appname <platformdirs.api.PlatformDirsABC.appname>`,
+    `appauthor <platformdirs.api.PlatformDirsABC.appauthor>`,
+    `version <platformdirs.api.PlatformDirsABC.version>`,
+    `roaming <platformdirs.api.PlatformDirsABC.roaming>`,
+    `opinion <platformdirs.api.PlatformDirsABC.opinion>`."""
+
+    @classmethod
+    def is_active(cls) -> bool:
+        """:return: a check to detect if Windows platform is currently active"""
+        return sys.platform == "win32"
+
+    @property
+    def user_data_dir(self) -> str:
+        """
+        :return: data directory tied to the user, e.g.
+         ``%USERPROFILE%\\AppData\\Local\\$appauthor\\$appname`` (not roaming) or
+         ``%USERPROFILE%\\AppData\\Roaming\\$appauthor\\$appname`` (roaming)
+        """
+        const = "CSIDL_APPDATA" if self.roaming else "CSIDL_LOCAL_APPDATA"
+        path = os.path.normpath(get_win_folder(const))
+        return self._path_with_app_version(path)
+
+    def _path_with_app_version(self, path: str, *, opinion_value: Optional[str] = None) -> str:
+        if self.appname:
+            assert self.appname is not None
```

Comment:
this seems redundant at first glance

Reply:
Hmmm, kept existing logic, but yeah this can be removed.